dashboards-observability/public/components/common/search/autocomplete.tsx

```diff
@@ -178,11 +180,14 @@ const getSuggestions = async (str: string, dslService: DSLService) => {
       currFieldType = fieldsFromBackend.find((field) => field.label === currField)?.type;
       return fullSuggestions.filter(({ label }) => label.startsWith(prefix) && prefix !== label);
     } else if (nextWhere === splittedModel.length - 2) {
-      return fillSuggestions(
-        str,
-        prefix,
-        await getDataValues(currIndex, currField, currFieldType, dslService)
-      );
+      if (isEmpty(prefix)) {
+        return fillSuggestions(
+          str,
+          prefix,
+          await getDataValues(currIndex, currField, currFieldType, dslService)
+        );
+      }
+      return [];
```

Comment:
Wouldn't this wipe the suggestions after the user types the first character of the datavalue?

Reply:
Previously it simply just kept sending requests to retrieve values for a field on every typing. The existing implementation would cause errors/issues since it needs to change to use keyword rather than text for search. I checked with Ani on this yesterday, he's asking to disable the call when user typed in at least one character. I think this is going to be just a quick fix to prevent from overwhelming the backend and having many errors on each typing. But I think we would change the implementation later.

Reply:
Previously it simply just kept sending requests to retrieve values for a field on every typing. The existing implementation would cause errors/issues since it needs to change to use keyword rather than text for search. I checked with Ani on this yesterday, he's asking to disable the call when user typed in at least one character. This is going to be just a quick fix to prevent from overwhelming the backend and having many errors on each typing. But I think we would change the implementation later.